packages/clay-core/src/tree-view/TreeView.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+import {DndProvider} from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+
+import {ChildrenFunction, Collection, ICollectionProps} from './Collection';
+import {TreeViewGroup} from './TreeViewGroup';
+import {TreeViewItem, TreeViewItemStack} from './TreeViewItem';
+import {Icons, TreeViewContext} from './context';
+import {ITreeProps, useTree} from './useTree';
+
+function generateItemIndices<T>(
```

Comment:
@matuzalemsteles I guess that this function could be inside (or part of) a custom hook.

Initially what I had done is something like this (in pseudo code)

```tsx
export function TreeView<T>({
        ...
	items: initialItems,
	...
}: ITreeViewProps<T>) {
   const [items, setItems] = React.useState(generateItemIndices(initialItems));
   ...
}
```

turns out that broke everything (nothing was rendered) and I couldn't figure out why so I went with this "ugly solution" just to have something working. A hook would probably be the good place to also "move" the items which is currently done in `TreeViewItem`. Apart from that, there's not much but I also noticed that when dropping items in some cases there's no update. If you have any ideas let me know.

Reply:
@julien yeah I'm taking a look at it now. Probably as you are manipulating the nested structure, this does not cause a rendering, the ideal would be to create a copy of the structure and then manipulate and save in state, I am thinking of some solutions now.